test/colony-network-recovery.js

```diff
@@ -0,0 +1,498 @@
+      let nNodes = await colonyNetwork.getReputationRootHashNNodes();
+      assert.equal(rootHash, "0x0100000000000000000000000000000000000000000000000000000000000000");
+      assert.equal(nNodes.toNumber(), 0);
+
+      await colonyNetwork.enterRecoveryMode();
+
+      await colonyNetwork.setStorageSlotRecovery(19, "0x02");
+      await colonyNetwork.setStorageSlotRecovery(20, `0x${new BN(7).toString(16, 64)}`);
+
+      await colonyNetwork.approveExitRecovery();
+      await colonyNetwork.exitRecoveryMode();
+      rootHash = await colonyNetwork.getReputationRootHash();
+      nNodes = await colonyNetwork.getReputationRootHashNNodes();
+      assert.equal(rootHash, "0x0200000000000000000000000000000000000000000000000000000000000000");
+      assert.equal(nNodes.toNumber(), 7);
+    });
+
+    process.env.SOLIDITY_COVERAGE // eslint-disable-line no-unused-expressions
+      ? it.skip
+      : it("miner should be able to correctly interpret historical reputation logs replaced during recovery mode", async () => {
+          await giveUserCLNYTokensAndStake(colonyNetwork, accounts[5], toBN(10).pow(toBN(18)));
+          await miningClient.saveCurrentState();
+          const startingHash = await miningClient.getRootHash();
+
+          const newMiningClient = new ReputationMiner({
+            loader: contractLoader,
+            minerAddress: accounts[5],
+            realProviderPort: REAL_PROVIDER_PORT,
+            useJsTree: true
+          });
+          await newMiningClient.initialise(colonyNetwork.address);
+
+          const tokenArgs = getTokenArgs();
+          const token = await Token.new(...tokenArgs);
+          const { logs } = await colonyNetwork.createColony(token.address);
+          const { colonyAddress } = logs[0].args;
+
+          await token.setOwner(colonyAddress);
+          const colony = await IColony.at(colonyAddress);
+
+          await colony.mintTokens(1000000000000000);
+          await colony.bootstrapColony([accounts[0]], [1000000000000000]);
+
+          let addr = await colonyNetwork.getReputationMiningCycle.call(false);
+          let repCycle = await IReputationMiningCycle.at(addr);
+
+          await miningClient.addLogContentsToReputationTree();
+          await forwardTime(3600, this);
+          await miningClient.submitRootHash();
+
+          addr = await colonyNetwork.getReputationMiningCycle.call(true);
+          repCycle = await IReputationMiningCycle.at(addr);
+          await repCycle.confirmNewHash(0);
+
+          await miningClient.addLogContentsToReputationTree();
+          await forwardTime(3600, this);
+          await miningClient.submitRootHash();
+
+          addr = await colonyNetwork.getReputationMiningCycle.call(true);
+          repCycle = await IReputationMiningCycle.at(addr);
+
+          const invalidEntry = await repCycle.getReputationUpdateLogEntry(5);
+          invalidEntry.amount = 0;
+
+          await repCycle.confirmNewHash(0);
```

Comment:
Here the currently active reputation log is destructed, the inactive becomes active and a blank inactive cycle is created.